File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.cc
#include <iostream>
#include <string>
#include <stack>
using namespace std;

void checkStr(string s) {
    bool containVowel = false; // 모음 여부
    bool throwFlag = false; // 불가능한 문자열인지 여부

    stack<char> vowel;
    stack<char> consonant;

    for (int i=0; i<s.length(); i++)
    {
        if (throwFlag) {
            break;
        }
        char curr = s[i];
        
        if (i > 0) {
            char prev = s[i-1]; // 이전 문자
            if (prev == curr) // 두번 연속 나오는 경우
                if (curr != 'e' && curr != 'o')
                    throwFlag = true;
        }

        if (curr == 'a' || curr == 'e' || curr == 'i' || curr == 'o' || curr == 'u') { // 모음 나옴
            containVowel = true;
            vowel.push(curr);
            if (vowel.size() == 3) // 모음 3개 연속
                throwFlag = true;
            else {
                while (!consonant.empty()) { // 자음 스택 clear
                    consonant.pop();
                }
            }
        } else {
            consonant.push(curr);
            if (consonant.size() == 3) // 자음 3개 연속
                throwFlag = true;
            else {
                while (!vowel.empty()) { // 모음 스택 clear
                    vowel.pop();
                }
            }
        }
    }
    
    if (!containVowel || throwFlag) {
        cout << "<" << s << "> is not acceptable.\n";
        return;
    } else {
        cout << "<" << s << "> is acceptable.\n";
        return;
    }
}

int main() {
    while (true) {
        string s;
        cin >> s;
        if (s == "end") {
            break;
        }

        checkStr(s);
    }
}

File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.py
import sys

def checkPassword(password):
  containVowel = False
  throwFlag = False

  vowel = []
  consonant = []

  for i in range(len(password)):
    curr = password[i]
    
    # 이전과 같은 글자 처리
    if (i != 0):
      if (curr == password[i-1]):
        if (curr not in ['e', 'o']):
          throwFlag = True

    # 모음/자음 연속 처리
    if (curr in ['a', 'e', 'i', 'o', 'u']): # 모음일때
      containVowel = True
      vowel.append(curr)
      if (len(vowel) >= 3): # 모음 3개 연속
        throwFlag = True
      else:
        consonant = []
    else: # 자음일때
      consonant.append(curr)
      if (len(consonant) >= 3):
        throwFlag = True
      else:
        vowel = []

  if (not containVowel or throwFlag): return False
  else: return True

while True:
  input = sys.stdin.readline().strip()
  if input == 'end':
    break
  if checkPassword(input):
    # print('<' + input.strip() + '> is acceptable.')
    print(f'<{input}> is acceptable.')
  else:
    # print('<' + input.strip() + '> is not acceptable.')
    print(f'<{input}> is not acceptable.')